### DISCUSSION TITLE
Example ViteNet.React not working with NuGet Package

### DISCUSSION THREAD

**Discussion by byte2pixel** [Eptagone/Vite.AspNetCore#110] [Q&A] [answered]
This is probably something really simple, but I can't figure it out.

I cloned the repo and can build and run the example ViteNet.React project just fine like that.

However, I was trying to see how to work with the NuGet package.

So I changed the ViteNet.React.csproj file and replaced the project reference (below)
```xml
  <ItemGroup>
    <ProjectReference Include="..\..\..\src\Vite.AspNetCore\Vite.AspNetCore.csproj" />
  </ItemGroup>
```
.. with the NuGet package instead of the project reference.  Using `dotnet add package Vite.AspNetCore` as the instructions say.  Everything still builds and runs fine but the page doesn't seem to render correctly.  Looking at the network / sources it seems everything is received but some things different for sure.  I definite saw the react refresh stuff missing when I switched to the NuGet package.

I'm just not sure what the catch is.

To recap all I did was swap the project reference out in place of the NuGet package.

Any ideas?

**Comment by byte2pixel** [comment]
I would like to add the following findings this morning.

Running the application from the bin\Release\net8.0 folders works fine
using something like:  `.\bin\Release\net8.0\ViteNET.React.exe --urls="https://localhost:7053;http://localhost:5044"`
The only time the example doesn't seem to work is when just using `dotnet run` from the ViteNet.React folder (while using the NuGet package)  when I swap the nuget package out for the project reference then  just using `dotnet run` works fine.

**Reply by byte2pixel** [reply]
I noticed this all has to do with the line in the example
```cs
if (app.Environment.IsDevelopment())
{
    // Use Vite Dev Server as middleware.
    app.UseViteDevelopmentServer();
}
```

If I comment out that line, then everything works when using the NuGet package.  However when using the dev server it seems to not work.  I see these errors in the console.

```
Building...
info: Microsoft.Hosting.Lifetime[14]
      Now listening on: https://localhost:7053
info: Microsoft.Hosting.Lifetime[14]
      Now listening on: http://localhost:5044
info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Development
info: Microsoft.Hosting.Lifetime[0]
      Content root path: D:\development\repos\Vite.AspNetCore\ViteNET.React
info: System.Net.Http.HttpClient.Vite.AspNetCore.DevHttpClient.LogicalHandler[100]
      Start processing HTTP request GET https://localhost:5173/
info: System.Net.Http.HttpClient.Vite.AspNetCore.DevHttpClient.ClientHandler[100]
      Sending HTTP request GET https://localhost:5173/
info: System.Net.Http.HttpClient.Vite.AspNetCore.DevHttpClient.LogicalHandler[100]
      Start processing HTTP request GET https://localhost:5173/
info: System.Net.Http.HttpClient.Vite.AspNetCore.DevHttpClient.ClientHandler[100]
      Sending HTTP request GET https://localhost:5173/
info: Vite.AspNetCore.Services.ViteDevServerStatus[0]
      Starting the vite development server...
info: Vite.AspNetCore.Services.ViteDevServerStatus[0]
      Vite development server started with process ID 35404.
info: System.Net.Http.HttpClient.Vite.AspNetCore.DevHttpClient.LogicalHandler[100]
      Start processing HTTP request GET https://localhost:5173/
info: System.Net.Http.HttpClient.Vite.AspNetCore.DevHttpClient.ClientHandler[100]
      Sending HTTP request GET https://localhost:5173/
// 404s
info: System.Net.Http.HttpClient.Vite.AspNetCore.DevHttpClient.ClientHandler[101]
      Received HTTP response headers after 531.2687ms - 404
info: System.Net.Http.HttpClient.Vite.AspNetCore.DevHttpClient.LogicalHandler[101]
      End processing HTTP request after 535.5323ms - 404
// end 404s
info: Vite.AspNetCore.Services.ViteDevServerStatus[0]
      The Vite development server is running at https://localhost:5173
info: Vite.AspNetCore.Services.ViteManifest[0]
      The manifest file won't be read because the vite development service is enabled. The service will always return null chunks
```

Additionally if I try to access the vite dev server directly using
`https://localhost:5173/`

I get 404:
Cannot GET /

I assume that is corresponding to the AspNet log  messages that say 404.

I was just trying to get this to work so HMR would be useable.

**Comment by Eptagone** [comment, marked as answer]
Hi, the NuGet package doesn't have the last changes yet. I'm refactoring the code before publishing the new version.

**Reply by byte2pixel** [reply]
Ah that explains it!  I thought I was doing something wrong for sure but couldn't figure out for the life of me what it was.

**Reply by byte2pixel** [reply]
Saw the update today.  Updated the package in my project and added `options.Server.UseReactRefresh = true;` to the AddViteServices action and now I am able to use `dotnet run` and HRM works great now.
